review-notifications/src/Options.js

```diff
@@ -119,92 +203,197 @@ class Options extends Component {
       'github.com',
       'api.github.com/repos'
     );
-    let newRepos = this.state.followedRepos;
-
-    axios
-      .get(apiLink)
-      .then(response => {
-        newRepos.push({
-          name: response.data.full_name,
-          link: this.state.newRepo,
-          prLink: response.data.pulls_url.replace('{/number}', ''),
-        });
-        this.setState({ followedRepos: newRepos, newRepo: '' }, () => {
-          chrome.storage.local.set({ followedRepos: this.state.followedRepos });
-          chrome.runtime.sendMessage({
-            message: 'Changed followed repositories',
+
+    if (
+      !this.state.followedRepos.find(repo => repo.link === this.state.newRepo)
+    ) {
+      let newRepos = this.state.followedRepos;
```

Comment:
this can be very dangerous. Lists and objects in javascript are mutable. What you are doing here is not copying the list value but passing it's reference. Consider this example:

![image](https://user-images.githubusercontent.com/36732892/56276836-8d019c80-6103-11e9-827e-4c64361db77d.png)

And as you probably already know, mutating the state directly is very dangerous. You can copy it by value by doing
`const newRepos = this.state.followedRepos.slice()`

Also newRepos should be a `const` instead of `let`

Reply:
I might have missed other instances of the code if you are assigning state like this, please check thoroughly as this is a very major mistake and can cause a lot of bugs.